core/commands/p2p.go

```diff
@@ -53,49 +56,121 @@ are refined`,
 	},
 
 	Subcommands: map[string]*cmds.Command{
-		"listener": p2pListenerCmd,
-		"stream":   p2pStreamCmd,
+		"stream": p2pStreamCmd,
+
+		"forward": p2pForwardCmd,
+		"close":   p2pCloseCmd,
+		"ls":      p2pLsCmd,
 	},
 }
 
-// p2pListenerCmd is the 'ipfs p2p listener' command
-var p2pListenerCmd = &cmds.Command{
+var p2pForwardCmd = &cmds.Command{
```

Comment:
There's also a third case: "connect". That is, if I'm listening on localhost:X, I may want to call `ipfs p2p connect myproto /ip4/localhost/tcp/X /ipfs/QmId`. I'd expect IPFS to:

1. Open a `myproto` stream to `/ipfs/QmId`.
2. Connect to `/ip4/localhost/tcp/X`.
3. Start forwarding the connections.
4. Return the local port (on the CLI/API) IPFS used when connecting to `/ip4/localhost/tcp/X`.
5. Close the stream when the client closes their end of the connection.

Users of existing applications will likely use the forward command but developers of new applications that want to puppet a remote ipfs daemon will likely want to use the `connect` command.

---

This can obviously be added later. I'm just motivating the multiple command option.